src/os/bluestore/BlueStore.cc

```diff
@@ -9015,7 +9015,210 @@ int BlueStore::_split_collection(TransContext *txc,
   return r;
 }
 
+#define KEY_SLAB 32
+#define VALUE_SLAB 64
 
+int get_key_slab(size_t sz)
+{
+  return (sz/KEY_SLAB);
+}
+
+string get_key_slab_to_range(int slab)
+{
+  int lower_bound = slab * KEY_SLAB;
+  int upper_bound = (slab + 1) * KEY_SLAB;
+  string ret = "[" + stringify(lower_bound) + "," + stringify(upper_bound) + ")";
+  return ret;
+}
+
+int get_value_slab(size_t sz)
+{
+  return (sz/VALUE_SLAB);
+}
+
+string get_value_slab_to_range(int slab)
+{
+  int lower_bound = slab * VALUE_SLAB;
+  int upper_bound = (slab + 1) * VALUE_SLAB;
+  string ret = "[" + stringify(lower_bound) + "," + stringify(upper_bound) + ")";
+  return ret;
+}
+
+//Itrerates through the db and collects the stats
+void BlueStore::get_db_histogram(Formatter *f)
+{
+  //globals
+  uint64_t num_onodes = 0;
+  uint64_t num_shards = 0;
+  uint64_t num_super = 0;
+  uint64_t num_coll = 0;
+  uint64_t num_omap = 0;
+  uint64_t num_wal = 0;
+  uint64_t num_alloc = 0;
+  uint64_t num_stat = 0;
+  uint64_t num_others = 0;
+  uint64_t num_shared_shards = 0;
+  size_t max_key_size =0, max_value_size = 0;
+  uint64_t total_key_size = 0, total_value_size = 0;
+  size_t key_size = 0, value_size = 0;
+  int key_slab, value_slab;
+
+  //histogram
+  struct value_dist
+  {
+    uint64_t count;
```

Comment:
@ifed01 these are part of the histogram, don't need to specifically initialize to 0, if the value doesn't exist in the map, it will be initialized to 0 by default.